fondant/dataset.py

```diff
@@ -166,82 +167,123 @@ def upload(self, save_path):
         return None
 
 
-class FondantComponent:
-    def __init__(self, type="transform"):
-        # note: Fondant spec always needs to be called like this
-        # and placed in the src directory
+class FondantComponent(ABC):
+    """Abstract base class for a Fondant component"""
+
+    def __init__(self):
         self.spec = FondantComponentSpec.from_file("fondant_component.yaml")
-        self.type = type
 
-    def run(self) -> dd.DataFrame:
-        """
-        Parses input data, executes the transform, and creates the output manifest.
+    @abstractmethod
+    def _get_manifest(self, args: argparse.Namespace) -> Manifest:
```

Comment:
Not a fan of defining abstract methods which are not called by the base class. It imposes a structure on the subclasses without any advantage to them.